.k8s_ci.Dockerfile

```diff
@@ -181,7 +189,7 @@ COPY --chown=$GALAXY_USER:$GALAXY_USER --from=client_build $SERVER_DIR/static ./
 WORKDIR $SERVER_DIR
 
 # The data in version.json will be displayed in Galaxy's /api/version endpoint
-RUN printf "{\n  \"git_commit\": \"$(cat GITREVISION)\",\n  \"build_date\": \"$BUILD_DATE\",\n  \"image_tag\": \"$IMAGE_TAG\"\n}\n" > version.json \
+RUN printf "{\n  \"git_commit\": \"$GIT_COMMIT\",\n  \"build_date\": \"$BUILD_DATE\",\n  \"image_tag\": \"$IMAGE_TAG\"\n}\n" > version.json \
```

Comment:
GIT_COMMIT is not guaranteed to be set?

Reply:
The reason it was done this way originally is that when you run `docker build . -t galaxy:latest` locally, the GIT_COMMIT won't be passed in, and there's no way to determine the commit after that.

Reply:
If someone builds locally then the BUILD_DATE and IMAGE_TAG should also be set, and if those are going to be set on the command line why not also set GIT_COMMIT?   
```
docker build . -t galaxy:latest --build-arg GIT_COMMIT=${git rev-parse HEAD) ...
```
That saves copying a ~1.3GB directory to the build context to get the value of one variable.

Reply:
The BUILD_DATE and IMAGE_TAG are useful but optional. The former can be determined from docker hub while the latter is inferable from the GIT_COMMIT. Hence why the GIT_COMMIT is what's really important. If removing .git from the build context makes a substantial enough improvement (what kind of difference do you see?), I'm ok with it, but we should at least document setting the GIT_COMMIT. What do you think @afgane?